File: Math/1360. Number of Days Between Two Dates.cpp
class Solution {
public:
    int days[12] = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
    int daysBetweenDates(string d1, string d2) {
        return abs(daysFrom1971(d1) - daysFrom1971(d2));
    }
    bool isLeap(int y) { 
        return y % 4 == 0 && (y % 100 != 0 || y % 400 == 0); 
    }
    int daysFrom1971(string dt) {
        int y = stoi(dt.substr(0, 4)), m = stoi(dt.substr(5, 2)), d = stoi(dt.substr(8));
        for (int iy = 1971; iy < y; ++iy) 
            d += isLeap(iy) ? 366 : 365;
        return d + (m > 2 && isLeap(y)) + accumulate(begin(days), begin(days) + m - 1, 0);
    }  
};

File: Math/1360. Number of Days Between Two Dates.py
class Solution:
    def daysBetweenDates(self, date1: str, date2: str) -> int:
        def f(date):
            year, month, day = [*date]
            #-1 是算之前一年的
            days = (year-1) * 365 + (year-1) // 4 - (year-1) // 100 + (year-1) // 400 
            for i in range(1, month):
                print(i, days)
                if i in set([1,3,5,7,8,10]):
                    days += 31
                elif i in set([4,6,9,11]):
                    days += 30
                elif year % 4 == 0 and year% 100 != 0 or year%400 == 0:
                        days += 29 
                else:
                    days += 28
            return days + day 
        return  abs(f(map(int, date1.split('-'))) - f(map(int, date2.split('-'))))
                